File: LSP+ISP/Spoilers/MarsRoverKata/Commands/BackwardCommand.cs
namespace MarsRoverKata.Commands
{
    public class BackwardCommand : ICommand
    {
        private Rover Rover { get; set; }

        public BackwardCommand(Rover rover)
        {
            Rover = rover;
        }

        public bool Execute()
        {
            return Rover.MoveBackward();
        }
    }
}

File: SRP/Java_Exercise/MarsRoverKata/src/main/java/MarsRoverKata/Commands/BackwardCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.Rover;

public class BackwardCommand implements ICommand {
    private Rover rover;

    public Rover getRover() {
        return rover;
    }

    public void setRover(Rover rover) {
        this.rover = rover;
    }

    public BackwardCommand(Rover movable) {
        this.rover = movable;
    }

    public boolean execute() {
        return rover.moveBackward();
    }
}

